go/mcap/indexed_message_iterator.go

```diff
@@ -144,9 +165,48 @@ func (it *indexedMessageIterator) parseSummarySection() error {
 			it.statistics = stats
 		case TokenFooter:
 			it.hasReadSummarySection = true
-			return nil
+			break lexerloop
+		}
+	}
+	it.chunksOverlap = false
+	switch it.indexHeap.order {
+	case FileOrder:
+		sort.Slice(it.chunkIndexesToLoad, func(i, j int) bool {
+			return it.chunkIndexesToLoad[i].ChunkStartOffset < it.chunkIndexesToLoad[j].ChunkStartOffset
+		})
+	case LogTimeOrder:
+		sort.Slice(it.chunkIndexesToLoad, func(i, j int) bool {
+			return it.chunkIndexesToLoad[i].MessageStartTime < it.chunkIndexesToLoad[j].MessageStartTime
+		})
+		for i := 1; i < len(it.chunkIndexesToLoad); i++ {
```

Comment:
iterating from the 1 seems a little unnatural IMO. Iterating from 0 to len(it.chunkIndexesToLoad)-1 seems a bit more natural, if there isn't something I'm missing.